src/huggingface_hub/repository.py

```diff
@@ -107,6 +112,98 @@ def is_git_ignored(filename: Union[str, Path]) -> bool:
+    """
+
+    def output_progress(stopping_event: threading.Event):
+        """
+        To be launched as a separate thread with an event meaning it should stop the tail.
+        """
+        pbars = {}
+
+        def close_pbars():
+            for pbar in pbars.values():
+                pbar["bar"].update(pbar["bar"].total - pbar["past_bytes"])
+                pbar["bar"].refresh()
+                pbar["bar"].close()
+
+        def tail_file(filename) -> Iterator[str]:
+            """
+            Creates a generator to be iterated through, which will return each line one by one.
+            Will stop tailing the file if the stopping_event is set.
+            """
+            with open(filename, "r") as file:
+                current_line = ""
+                while True:
+                    if stopping_event.is_set():
+                        close_pbars()
+                        break
+
+                    line_bit = file.readline()
+                    if line_bit is not None and not len(line_bit.strip()) == 0:
+                        current_line += line_bit
+                        if current_line.endswith("\n"):
+                            yield current_line
+                            current_line = ""
+                    else:
+                        time.sleep(1)
+
+        # If the file isn't created yet, wait for a few seconds before trying again.
+        # Can be interrupted with the stopping_event.
+        while not os.path.exists(os.environ["GIT_LFS_PROGRESS"]):
+            if stopping_event.is_set():
+                close_pbars()
+                break
+
+            time.sleep(2)
+
+        for line in tail_file(os.environ["GIT_LFS_PROGRESS"]):
+            state, file_progress, byte_progress, filename = line.split()
+            description = f"{state.capitalize()} file {filename}"
+
+            current_bytes, total_bytes = byte_progress.split("/")
+
+            current_bytes = int(current_bytes)
+            total_bytes = int(total_bytes)
+
+            if pbars.get((state, filename)) is None:
+                pbars[(state, filename)] = {
+                    "bar": tqdm(
+                        desc=description,
+                        initial=current_bytes,
+                        total=total_bytes,
+                        unit="B",
+                        unit_scale=True,
+                        unit_divisor=1024,
+                    ),
+                    "past_bytes": current_bytes,
+                }
+            else:
+                past_bytes = pbars[(state, filename)]["past_bytes"]
+                pbars[(state, filename)]["bar"].update(current_bytes - past_bytes)
+                pbars[(state, filename)]["past_bytes"] = current_bytes
+
+    current_lfs_progress_value = os.environ.get("GIT_LFS_PROGRESS", "")
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        os.environ["GIT_LFS_PROGRESS"] = tmpdir + "/lfs_progress"
```

Comment:
Will the "/" work on any OS (Windows)?

Reply:
Correct! I'll update to `os.path.join`. Thank you!